libs/ui/src/components/rich-input.tsx

```diff
@@ -113,25 +113,29 @@ const InsertImageForm = ({ onInsert }: InsertImageProps) => {
   );
 };
 
-const Toolbar = ({ editor }: { editor: Editor }) => {
-  const setLink = useCallback(() => {
-    const previousUrl = editor.getAttributes("link").href;
-    const url = window.prompt("URL", previousUrl);
+const setLinkGlobal = (editor: Editor) => {
```

Comment:
I'm open to changing the naming of this function if desired, I just decided to leave `Toolbar.setLink` as-is to keep the `diff` of this PR low for now.

Reply:
I actually don't mind if the code is refactored to be inside Toolbar, and within a useCallback. Would be neater that way. I can take it from here and make the necessary changes.

Thank you for this quick helpful fix.